Add test to sync a repository in Pulp3.

kersommoura opened the pull request:
In order to sync a repository an importer has to be associated within
this repository.  When a repository is created this version field is set
as None. After a sync the repository version is updated.

Do the following:

 1. Create a repository, and an importer.
 2. Assert that repository version is None.
 3. Sync the importer.
 4. Assert that repository version is not None.
 5. Sync the importer one more time.
 6. Assert that repository version is different from the previous one.

See: #844

Ichimonji10 commented on docs/api.rst:
```diff
@@ -131,7 +131,8 @@ developers, not a gospel.
     api/pulp_smash.tests.pulp3.file.api_v3
     api/pulp_smash.tests.pulp3.file.api_v3.test_crud_importers
-    api/pulp_smash.tests.pulp3.file.api_v3.test_crud_publishers
+    api/pulp_smash.tests.pulp3.file.api_v3.test_crud_publishers 
```
Please drop the whitespace from the end of this line.

Ichimonji10 commented on pulp_smash/tests/pulp3/utils.py:
```diff
@@ -2,6 +2,7 @@
 import random
 import unittest
+from urllib.parse import urljoin
 from copy import deepcopy
```
Please sort imports.

```python
import random
import unittest
from copy import deepcopy
from urllib.parse import urljoin
```

Ichimonji10 commented on pulp_smash/tests/pulp3/utils.py:
```diff
@@ -157,3 +158,21 @@ def get_plugins(cfg=None):
+
+    :returns: The server's response. Call ``.json()`` on the response to get a
+         call report.
+
```
Please drop the extra whitespace after each `:param:`, etc.

Ichimonji10 commented on pulp_smash/tests/pulp3/file/api_v3/test_sync.py:
```diff
@@ -0,0 +1,51 @@
+
+
+class SyncFileRepoTestCase(unittest.TestCase):
+    """Test that sync file plugin repositories."""
```
The wording here is a bit strange. Maybe you mean something like:

> Test that repositories can be synced with the file plugin.

Or:

> Sync repositories with the file plugin.

Ichimonji10 commented:
How about extending the test, so that a second sync is performed? The test case can assert that `repo['_latest_version_href']` has changed.

kersommoura commented:
@Ichimonji10, I addressed your comments, and I can expand the tests as you suggested.

Besides that, I was thinking to explore what is described here. 

[Sync of unchanged importer update repo version](https://pulp.plan.io/issues/3308)

Ichimonji10 commented:
This should work:

```python
# Sync the repository.
self.assertEqual(repo['_latest_version_href'], None)
sync_repo(cfg, importer)
repo = client.get(repo['_href'])
self.assertNotEqual(repo['_latest_version_href'], None)

# Sync the repository again.
latest_version_href = repo['_latest_version_href']
sync_repo(cfg, importer)
repo = client.get(repo['_href'])
self.assertNotEqual(latest_version_href, repo['_latest_version_href'])
```

kersommoura commented:
Test was updated per last comments. Thank you.

Ichimonji10 commented:
Rebased and merged as 144be870a0c77c6e7d0c3f16687ae318ce2a92df, with a follow-up commit of 064f59a35b65a03f7ce85abbf5b5f0863529fdee. Good job! Also see: [Support On_demand and Background downloaders](https://pulp.plan.io/issues/3320).
